homeassistant/helpers/entity_component.py

```diff
@@ -384,9 +385,12 @@ def _update_entity_states(self, now):
                     to_update.append(update_coro)
 
             for update_coro in to_update:
-                yield from update_coro
+                try:
+                    yield from update_coro
+                except Exception:  # pylint: disable=broad-except
+                    self.component.logger.exception(
+                        'Error while update entity in %s',
+                        self.component.domain)
 
             if tasks:
                 yield from asyncio.wait(tasks, loop=self.component.hass.loop)
```

Comment:
We should never rely on the exception handler. The exception handler is for exceptions that we did not expect.  Here we kinda expect exceptions from time to time.